xbmc/GUIInfoManager.cpp

```diff
@@ -5926,7 +5926,15 @@ bool CGUIInfoManager::GetItemBool(const CGUIListItem *item, int condition) const
       if (pItem->HasEPGInfoTag())
       {
         CFileItemPtr timer = g_PVRTimers->GetTimerForEpgTag(pItem);
-        if (timer && timer->HasPVRTimerInfoTag() && timer->GetPVRTimerInfoTag()->GetTimerScheduleId() != PVR_TIMER_NO_PARENT)
+        return (timer && timer->HasPVRTimerInfoTag() && timer->GetPVRTimerInfoTag()->GetTimerScheduleId() != PVR_TIMER_NO_PARENT);
+      }
+    }
+    else if (condition == LISTITEM_ISENABLED)
+    {
+      if (pItem->HasEPGInfoTag())
+      {
+        CFileItemPtr timer = g_PVRTimers->GetTimerForEpgTag(pItem);
+        if (timer && timer->HasPVRTimerInfoTag())
           return timer->GetPVRTimerInfoTag()->IsActive();
```

Comment:
This will show 'Broken' recordings (PVR_TIMER_STATE_CONFLICT_NOK or PVR_TIMER_STATE_ERROR) as perfectly working recordings. I know this is what the code does at present, but we can do better!

Suggest:
```return (timer->GetPVRTimerInfoTag()->IsActive() && !timer->GetPVRTimerInfoTag()->IsBroken());```

Even better would be three icons:
1) 'disabled' timers (maybe something with dashed/grey edges and no red)
2) the one you propose (or better @ronie 's version) just for 'broken' timers.
3) the original one for 'will record' timers.